src/main/jbake/content/blog/2017/07/grafana-new-query-interface.adoc

```diff
@@ -0,0 +1,174 @@
+image::../../../../assets/img/blog/2017/2017-07-20-grafana-tag_operator.png[Selecting tag operator]
+endif::[]
+
+[.text-center]
+_Selecting the tag operator_
+
+[.text-center]
+ifndef::env-github[]
+image::/img/blog/2017/2017-07-20-grafana-tag_value.png[Selecting tag value]
+endif::[]
+ifdef::env-github[]
+image::../../../../assets/img/blog/2017/2017-07-20-grafana-tag_value.png[Selecting tag value]
+endif::[]
+
+[.text-center]
+_Selecting the tag value_
+
+All these selections are facilitated with dynamic suggestions, but you can still use Grafana template variables within tag values, the text field allows free typing.
+Once you've set up a tag query expression, the relevant metrics immediately show up on the chart and the list of available metrics in the dropdown list in updated.
+
+
+[.text-center]
+ifndef::env-github[]
+image::/img/blog/2017/2017-07-20-grafana-tag-filtering.png[Filtered metrics]
+endif::[]
+ifdef::env-github[]
+image::../../../../assets/img/blog/2017/2017-07-20-grafana-tag-filtering.png[Filtered metrics]
+endif::[]
+
+[.text-center]
+_Filtered metrics_
+
+This query builder should allow you to build almost any tag query that the Hawkular server understands.
+There's however some corner cases. For now, this builder doesn't allow to prioritize expressions with parenthesis.
+For instance, you cannot build `c1 = 'foo' OR (b1 != 'B' AND a1 = 'abcd')`. Fortunately, you can turn off the query builder and directly
+type in query expression:
+
+[.text-center]
+ifndef::env-github[]
+image::/img/blog/2017/2017-07-20-grafana-editor-mode.png[Toggle editor mode]
+endif::[]
+ifdef::env-github[]
+image::../../../../assets/img/blog/2017/2017-07-20-grafana-editor-mode.png[Toggle editor mode]
+endif::[]
+
+[.text-center]
+_Toggle editor mode_
+
+It will be sent as is to the Hawkular Metrics server.
+This will also be useful to fill the gap if the language evolves server-side and this plugin isn't updated immediately.
+
+=== Stats query
+
+The other important feature is the ability to run link:http://www.hawkular.org/hawkular-metrics/docs/user-guide/#_downsampling[stats] queries against Hawkular Metrics, instead of link:http://www.hawkular.org/hawkular-metrics/docs/user-guide/#_raw_data[raw] queries.
+There's several reasons to do that:
+
+- it reduces the network load, and client-side processing load, especially when raw data would contain tons of datapoints
+- it enables some aggregation methods
+- it also allows higher-level analysis with stats such as percentiles
+
+To enable it, just untick the _raw_ checkbox.
+
+[.text-center]
+ifndef::env-github[]
+image::/img/blog/2017/2017-07-20-grafana-stats-mode.png[Toggle stats mode]
+endif::[]
+ifdef::env-github[]
+image::../../../../assets/img/blog/2017/2017-07-20-grafana-stats-mode.png[Toggle stats mode]
+endif::[]
+
+[.text-center]
+_Toggle stats mode_
+
+Here you can set up _Multiple series aggregation_ to _None_, _Sum_ or _Average_, and _Stat type_ as a list of statistics among
```

Comment:
"Here"  needs to be clearer.  

Suggestion - When you clear the Raw checkbox, you can configure _Multiple series aggregation_ to _None_, _Sum_ or _Average_ and can configure _Stat type_ as _avg_, _min_, _max_, _median_, _sum_ or different percentiles.  You can display several different Stat Types within the same query.

(Might give a more complex example here so that people can see what a query with different stat types would look like?)